src/MSBuild/XMake.cs

```diff
@@ -782,6 +789,42 @@ private static void DebuggerLaunchCheck()
                     {
                         ReplayBinaryLog(projectFile, loggers, distributedLoggerRecords, cpuCount);
                     }
+                    else if ((getProperty.Length > 0 || getItem.Length > 0 || getTargetResult.Length > 0) && FileUtilities.IsSolutionFilename(projectFile))
+                    {
+
+                        CommandLineSwitchException.Throw("SolutionBuildInvalidForCommandLineEvaluation",
+                            getProperty.Length > 0 ? "getProperty" :
+                            getItem.Length > 0 ? "getItem" :
+                            "getTargetResult");
+                    }
+                    else if ((getProperty.Length > 0 || getItem.Length > 0) && (targets is null || targets.Length == 0))
+                    {
+                        try
+                        {
+                            Project project = Project.FromFile(projectFile, new Definition.ProjectOptions()
+                            {
+                                GlobalProperties = globalProperties,
+                                ToolsVersion = toolsVersion,
+                            });
+
+                            // Special case if the user requests exactly one property: skip json formatting
+                            if (getProperty.Length == 1 && getItem.Length == 0)
+                            {
+                                Console.WriteLine(project.GetPropertyValue(getProperty[0]));
```

Comment:
Doing something different would be unexpected to me from and MSBuild perspective, but I'll admit that it would be...surprising...for it to print out nothing and act like everything is peachy. Thoughts @baronfel?

Reply:
Works for me.